File: 소수 찾기.js
function solution(numbers) {
  var answer = [];
  let nums = numbers.split("");

  const isPrimeNum = (num) => {
    if (num <= 1) return false;
    for (let i = 2; i * i <= num; i++) {
      if (num % i === 0) return false;
    }
    return true;
  };

  const getPermutation = (arr, fixed) => {
    if (arr.length >= 1) {
      for (let i = 0; i < arr.length; i++) {
        const newNum = fixed + arr[i];
        const copyArr = [...arr];
        copyArr.splice(i, 1);
        if (!answer.includes(+newNum) && isPrimeNum(+newNum)) {
          answer.push(+newNum);
        }
        getPermutation(copyArr, newNum);
      }
    }
  };

  getPermutation(nums, "");
  return answer.length;
}

File: 수학/소수 찾기.py
n = int(input())
arr = list(map(int, input().split()))

answer = 0
for x in arr:
    if x == 1:
        continue
    elif x < 4:
        answer += 1
    else:
        for i in range(2, int(x**0.5) + 1):
            if x % i == 0:
                # 소수가 아님
                break
        else:
            answer += 1
print(answer)
